3.3/debian-10/rootfs/opt/bitnami/scripts/libetcd.sh

```diff
@@ -306,19 +306,21 @@ etcdctl_auth_flags() {
 ########################
 etcd_store_member_id() {
     local -a extra_flags
-
+    local member_id
     info "Obtaining cluster member ID"
     etcd_start_bg
     read -r -a extra_flags <<<"$(etcdctl_auth_flags)"
     is_boolean_yes "$ETCD_ON_K8S" && extra_flags+=("--endpoints=$(etcdctl_get_endpoints)")
     if retry_while "etcdctl ${extra_flags[*]} member list" >/dev/null 2>&1; then
-        while [[ ! -s "${ETCD_DATA_DIR}/member_id" ]]; do
-            # We use 'stdbuf' to ensure memory buffers are flushed to disk
-            # so we reduce the chances that the "member_id" file is empty.
-            # ref: https://www.gnu.org/software/coreutils/manual/html_node/stdbuf-invocation.html#stdbuf-invocation
+        while is_empty_value "${member_id}"; do
             read -r -a advertised_array <<<"$(tr ',;' ' ' <<<"$ETCD_ADVERTISE_CLIENT_URLS")"
-            stdbuf -oL etcdctl "${extra_flags[@]}" member list | grep -w "${advertised_array[0]}" | awk -F "," '{ print $1}' >"${ETCD_DATA_DIR}/member_id" || true
+            member_id=etcdctl "${extra_flags[@]}" member list | grep -w "${advertised_array[0]}" | awk -F "," '{ print $1}' || true
```

Comment:
We usually make use of sub-shells to assign values to existing variables. IMHO, it is easier to read.

```suggestion
            member_id="$(etcdctl "${extra_flags[@]}" member list | grep -w "${advertised_array[0]}" | awk -F "," '{ print $1}' || true)"
```